services/tables/src/main/java/com/linkedin/openhouse/tables/api/validator/impl/PoliciesSpecValidator.java

```diff
@@ -40,55 +41,106 @@ public class PoliciesSpecValidator {
   public boolean validate(
       Policies policies, TimePartitionSpec timePartitioning, TableUri tableUri, String schema) {
 
-    if (policies != null && policies.getRetention() != null) {
-      // Two invalid case for timePartitioned table
-      if (timePartitioning != null) {
-        if (policies.getRetention().getColumnPattern() != null) {
+    if (policies != null) {
+      if (policies.getRetention() != null) {
+        // Two invalid case for timePartitioned table
+        if (timePartitioning != null) {
+          if (policies.getRetention().getColumnPattern() != null) {
```

Comment:
I think retention policies validation can be done separately to keep the scope only to replication changes.

Reply:
Or is this a refactoring to incorporate replication policies validation?

Reply:
this was a refactor to incorporate replication policies validation, but I can make a new method for replication policies and leave this one for retention only 👍